tokio/src/process/mod.rs

```diff
@@ -860,8 +860,97 @@ impl Command {
     /// running on the system).
+    #[inline]
     pub fn spawn(&mut self) -> io::Result<Child> {
-        imp::spawn_child(&mut self.std).map(|spawned_child| Child {
+        // On two lines to circumvent a mutable borrow check failure.
+        let child = self.std.spawn()?;
+        self.build_child(child)
+    }
+
+    /// Executes the command as a child process with a custom spawning function,
+    /// returning a handle to it.
+    ///
+    /// This is identical to [`Self::spawn`] in every aspect except the spawn:
+    /// here, it is customizable through the `with` parameter instead of
+    /// defaulting to the usual spawn. In fact, [`Self::spawn`] is just
+    /// [`Self::spawn_with`] with [`StdCommand::spawn`].
+    ///
+    /// This is useful mostly under Windows for now, since the platform exposes
+    /// special APIs to configure child processes when spawning them with various
+    /// attributes that customize the exact behavior of the spawn operation.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```no_run
+    /// # async fn test() { // allow using await
+    /// use std::process::Stdio;
+    ///
+    /// let output = tokio::process::Command::new("ls")
+    ///         .stdin(Stdio::null())
+    ///         .stdout(Stdio::piped())
+    ///         .stderr(Stdio::piped())
+    ///         .spawn_with(std::process::Command::spawn)
+    ///         .unwrap()
+    ///         .wait_with_output()
+    ///         .await
+    ///         .unwrap();
+    /// # }
+    /// ```
+    ///
+    /// Actually customizing the spawn under Windows:
+    ///
+    /// ```ignore
+    /// # #[cfg(windows)]   // Windows-only nightly APIs are used here.
+    /// # async fn test() { // Allow using await.
+    /// use std::os::windows::io::AsRawHandle;
+    /// use std::os::windows::process::{CommandExt, ProcThreadAttributeList};
+    /// use std::process::Stdio;
+    /// use tokio::process::Command;
+    ///
+    /// let parent = Command::new("cmd").spawn().unwrap();
+    /// let parent_process_handle = parent.as_raw_handle();
+    ///
+    /// const PROC_THREAD_ATTRIBUTE_PARENT_PROCESS: usize = 0x00020000;
+    /// let mut attribute_list = ProcThreadAttributeList::build()
+    ///     .attribute(PROC_THREAD_ATTRIBUTE_PARENT_PROCESS, &parent_process_handle)
+    ///     .finish()
+    ///     .unwrap();
+    ///
+    /// let output = Command::new("ls")
+    ///         .stdin(Stdio::null())
+    ///         .stdout(Stdio::piped())
+    ///         .stderr(Stdio::piped())
+    ///         .spawn_with(|cmd| cmd.spawn_with_attributes(&attribute_list))
+    ///         .unwrap()
+    ///         .wait_with_output()
+    ///         .await
+    ///         .unwrap();
+    /// # }
+    /// ```
+    #[cfg(tokio_unstable)]
+    #[inline]
+    pub fn spawn_with(
+        &mut self,
+        with: impl Fn(&mut StdCommand) -> io::Result<StdChild>,
+    ) -> io::Result<Child> {
```

Comment:
This is missing the "Available on `tokio_unstable` only" marker in the docs.
```rs
#[cfg_attr(docsrs, doc(cfg(tokio_unstable)))]
```

Reply:
Woops, thanks! Done.